src/features/charts/redux/middleware.ts

```diff
@@ -0,0 +1,15 @@
+import { Middleware, AnyAction } from 'redux'
+import { UPDATE_CHART_RANGE } from '../redux'
+
+export const chartMiddleware = (socket: SocketIOClient.Socket): Middleware => {
```

Comment:
You should add a generic to the `Middleware` type to tell it the store shape, for when `getState` is called.

Reply:
Implemented for middleware utilising getState!